src/register.jl

```diff
@@ -637,6 +637,7 @@ function register(
     registry = GitTools.normalize_url(registry)
     registry_repo = get_registry(registry; gitconfig=gitconfig, force_reset=force_reset, cache=cache)
     registry_path = LibGit2.path(registry_repo)
+    registry_default_branch = LibGit2.branch(registry_repo)
```

Comment:
@GunnarFarneback removed the search for `HEAD` here with a call to just `branch` from `LibGit2` which should return the default branch of the registry here.